PokeOneWeb/Frontend/pokeoneweb/src/app/pages/pokemon-detail/pokemon-detail.component.ts

```diff
@@ -167,7 +171,83 @@ export class PokemonDetailComponent implements OnInit {
         this.model.spawnsSortedByColumn = sortColumn;
         this.model.spawnsSortDirection = sortDirection;
 
-        this.model.pokemon.spawns = this.sortService.sortSpawns(this.model.pokemon.spawns, sortColumn, sortDirection);
+        this.model.visibleSpawns = this.sortService.sortSpawns(this.model.visibleSpawns, sortColumn, sortDirection);
+    }
+
+    public hideEventExclusiveSpawns() {
+        if (!this.model.pokemon) return;
+        
+        this.model.areEventExclusiveSpawnsHidden = true;
+        this.model.visibleSpawns = [];
+
+        this.checkAreNoEventSpawnsAvailable(this.model.pokemon.spawns);
+
+        for (let spawn of this.model.pokemon.spawns){
+            if(spawn.isEvent){
```

Comment:
I would move the spawn.isEvent check to the extracted method, so that you only have the for loop and a single if-statement in here. Negate the check in the method, i.e.
```
if (!spawn.isEvent) {
  return true;
}

// Check event spawn
```